gauntlet/packages/gauntlet-solana/src/commands/internal/solana.ts

```diff
@@ -107,7 +146,39 @@ export default abstract class SolanaCommand extends WriteCommand<TransactionResp
     }
     const signedTx = await this.wallet.signTransaction(tx)
     logger.loading('Sending tx...')
-    return await sendAndConfirmRawTransaction(this.provider.connection, signedTx.serialize())
+    try {
+      return await sendAndConfirmRawTransaction(this.provider.connection, signedTx.serialize())
+    } catch (error) {
+      // Retry mechanism with greater priority fees
+      if (error instanceof SendTransactionError && error.message.includes('congestion')) {
+        validateRetryPriorityInput(this.flags.priorityRetry)
+        const [percentBump, numberOfRetries] = this.flags.priorityRetry.split(',')
+        overrides.price *= 1 + percentBump
+        logger.info(
+          `Attempt: ${retryCount} - Transaction Failed due to network congestion, increasing by 
+          ${parseFloat(1 + percentBump) * 100}% with ${overrides.price} micro Lamports priority fee`,
+        )
+        if (retryCount < numberOfRetries) {
+          return this.signAndSendRawTx(rawTxs, extraSigners, overrides, (retryCount += 1))
+        }
+        throw error
+      } else if (error instanceof TransactionExpiredTimeoutError) {
+        // Sometimes it takes longer to confirm or we need to retry check the transaction
+        // Do 3 retries
+        const signature = error.signature
+        for (let i = 0; i < 3; i++) {
+          const status = await this.provider.connection.getSignatureStatus(signature)
+          if (status.value && status.value.confirmationStatus == 'confirmed') {
+            return signature
+          }
+          // exponential
+          this.sleep(3000 ** i)
+        }
+        throw error
```

Comment:
it's still possible however, for a transaction to not be picked up if the priority fee is too low. it's like on EVM if the transaction is stuck in the mempool, but you bump the fee to get it prioritized. in solana, the transaction would just expire though. 

to solve also your case, then we can postpone the priority retry after you've verified that the transaction won't be confirmed? so on line 177